com.unity.render-pipelines.universal/Editor/AssetPostProcessors/ShaderGraphMaterialsUpdater.cs

```diff
@@ -50,25 +41,14 @@ static void OnShaderGraphSaved(Shader shader, object saveContext)
                             i / (float)(length - 1));
                     }
 
-                    // Get Material object
-                    string materialPath = AssetDatabase.GUIDToAssetPath(materialGuids[i]);
-                    Material material = AssetDatabase.LoadAssetAtPath<Material>(materialPath);
-
                     // Reset keywords
-                    if ((material != null) && material.shader.name == shader.name)
-                        ShaderUtils.UpdateMaterial(material, ShaderUtils.MaterialUpdateType.ModifiedShader);
-
-                    material = null;
-
-                    // Free the materials every 200 iterations, on big project loading all materials in memory can lead to a crash
-                    if ((i % 200 == 0) && i != 0)
-                        EditorUtility.UnloadUnusedAssetsImmediate(true);
+                    if (materials[i].shader.name == shader.name)
+                        ShaderUtils.UpdateMaterial(materials[i], ShaderUtils.MaterialUpdateType.ModifiedShader);
```

Comment:
Why do we need this logic? I just tested and even on shader graph save, depending materials gets their ValidateMaterial message called.

Reply:
we do need it to reload materials in memory. There are a few things that can trigger a reset so maybe it was one of them, like if you have the material inspector openned.
I tested to disable the code and the material don't get reset